docs/book/basics/simple_controller.md

```diff
@@ -1,144 +1,139 @@
+
+> pkg/controller/containerset/containerset_controller.go
 
 ## Setup
 
 {% method %}
-
-Code generation requires the following to be defined in controller.go:
- 
-- a `ProvideController` function returning an initialized `controller.GenericController`.  This
-  will be called from the controller-manager at runtime.
-- a go struct annotated with `// +kubebuilder:controller`.  This wires the controller in the `inject` package.
-
 #### ContainerSetController
 
-ContainerSetController has 3 annotations:
+ContainerSetController has a single annotation:
 
-- `// +kubebuilder:controller` registers the controller in the generated `pkg/inject` code ensuring it will be
-  started by the controller-manager main at runtime.
-- `// +kubebuilder:rbac` creates RBAC rules in the `hack/install.yaml` file created by `kubebuilder create config`.
+- `// +kubebuilder:rbac` creates RBAC rules in the `config/rbac/group_kind.yaml` files when `make` is run.
   This will ensure the Kubernetes ServiceAccount running the controller can read / write to the Deployment API
-- `// +kubebuilder:informers` starts the informer for listening to Deployment events
 
 ContainerSetController has 2 variables:
 
-- `InjectArgs` contains the clients provided to *ProvideController*
-- `containersetrecorder` contains a recorder to publish information displayed by `kubectl describe`.
+- `client.Client` is a client for reading / writing Kubernetes APIs.
+- `scheme *runtime.Scheme` is a runtime.Scheme used by the library to set OwnerReferences.
+
+#### Add
+
+Add creates a new Controller that will be started by the Manager.  When adding a Controller it is important to setup
+Watch functions to trigger Reconciles.
+
+Watch is a function that takes an event `source.Source` and a `handler.EventHandler`.  The Source provides events
+for some type, and the EventHandler responds to events by enqueuing `reconcile.Request`s for objects.
+Watch optionally takes a list of Predicates that may be used to filter events.
 
-#### ProvideController
+Sources
 
-ProviderController configures a new controller to watch for changes and call Reconcile.
+- To watch for create / update / delete events for an object use a `source.KindSource` e.g.
+`source.KindSource{Type: &v1.Pod}`
 
-ProvideController will be called from `pkg/inject/zz_generated.kubebuilder.go` after running `kubebuider generate`.
+Handlers
 
-- Create a new `ContainerSetController` struct
-- Create a new `GenericController` with the `ContainerSetController` Reconcile function
-- Watch for events on *ContainerSet*s and call Reconcile for the key
-- Watch for events on *Deployment*s and call Reconcile for the key of the Owning ContainerSet
-- Return the `GenericController` from the function
+- To enqueue a Reconcile for the object in the event use a `handler.EnqueueRequestForObject`
+- To enqueue a Reconcile for the object that created the object in the event use a `handler.EnqueueRequestForOwner`
```

Comment:
Maybe `the owner object that created the object in the event`

Reply:
Done